packages/@glimmer/core/test/non-interactive/render-test.ts

```diff
@@ -0,0 +1,291 @@
+
+    class MyComponent extends Component {}
+    setComponentTemplate(
+      MyComponent,
+      createTemplate({ OtherComponent }, `<h1><OtherComponent /></h1>`)
+    );
+
+    const html = await render(MyComponent);
+    assert.strictEqual(html, '<h1>Hello world</h1>', 'the template was rendered');
+    assert.ok(true, 'rendered');
+  });
+
+  test('a component can render multiple nested components', async assert => {
+    class Foo extends Component {}
+    setComponentTemplate(Foo, createTemplate(`Foo`));
+
+    class Bar extends Component {}
+    setComponentTemplate(Bar, createTemplate(`Bar`));
+
+    class OtherComponent extends Component {}
+    setComponentTemplate(
+      OtherComponent,
+      createTemplate({ Foo, Bar }, `Hello world <Foo /><Bar />`)
+    );
+
+    class MyComponent extends Component {}
+    setComponentTemplate(
+      MyComponent,
+      createTemplate({ OtherComponent }, `<h1><OtherComponent /></h1>`)
+    );
+
+    const html = await render(MyComponent);
+
+    assert.strictEqual(html, '<h1>Hello world FooBar</h1>', 'the template was rendered');
+  });
+
+  test('custom elements are rendered', async function(assert) {
+    const component = templateOnlyComponent();
+
+    setComponentTemplate(component, createTemplate('<hello-world>foo</hello-world>'));
+
+    assert.equal(await render(component), '<hello-world>foo</hello-world>');
+  });
+
+  test('a component can render with helpers', async assert => {
+    const myHelper = helper(([name], { greeting }) => {
+      return `helper ${greeting} ${name}`;
+    });
+
+    class MyComponent extends Component {}
+    setComponentTemplate(
+      MyComponent,
+      createTemplate({ myHelper }, '<h1>{{myHelper "foo" greeting="Hello"}}</h1>')
+    );
+
+    const html = await render(MyComponent);
+    assert.strictEqual(html, '<h1>helper Hello foo</h1>', 'the template was rendered');
+  });
+
+  test('a component can render with args', async assert => {
+    class MyComponent extends Component {}
+
+    setComponentTemplate(MyComponent, createTemplate('<h1>{{@say}}</h1>'));
+
+    const renderOptions = {
+      args: {
+        say: 'Hello Dolly!',
+      },
+    };
+
+    const html = await render(MyComponent, renderOptions);
+    assert.strictEqual(
+      html,
+      '<h1>Hello Dolly!</h1>',
+      'the component is rendered with passed in args'
+    );
+  });
+
+  test('can use block params', async function(assert) {
+    class MainComponent extends Component {
+      salutation = 'Glimmer';
+    }
+
+    const HelloWorld = templateOnlyComponent();
+
+    setComponentTemplate(HelloWorld, createTemplate('{{yield @name}}!'));
+
+    setComponentTemplate(
+      MainComponent,
+      createTemplate(
+        { HelloWorld },
+        '<HelloWorld @name={{this.salutation}} as |name|>{{name}}</HelloWorld>'
+      )
+    );
+
+    assert.equal(await render(MainComponent), 'Glimmer!');
+  });
+
+  // [true, false].forEach((pred) => {
```

Comment:
Delete?

Reply:
Same as above, these will be valid tests added back in followups.